platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/DevelocityBuildLifecycleService.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise;
+
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.internal.service.scopes.Scopes;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+/**
+ * Features for Develocity used during settings configuration to register lifecycle callbacks.
+ *
+ * @since 8.8
+ */
+@ServiceScope(Scopes.Gradle.class)
+public interface DevelocityBuildLifecycleService {
```

Comment:
Good point. If this interface will only contain the one method that replaces `allprojects`/`beforeProject`, then I'd go with `DevelocityProjectConfigurationService`. I went with `BuildLifecycle` service since it exposes on lifecycle callback, `beforeProject`, and might expose more build lifecycle callbacks later.

I am fine to go with `DevelocityProjectConfigurationService`.

Reply:
The public API for the functionality I want to expose here will be `gradle.lifecycle.beforeProject`. The expectation is that isolated project compatible variants of `afterProject` and `buildFinished` will go to `gradle.lifecycle` as well.
In the light of that, does `DevelocityBuildLifecycleService` make sense?